src/components/input/SelectNext.tsx

```diff
@@ -0,0 +1,195 @@
+import classnames from 'classnames';
+import type { ComponentChildren } from 'preact';
+import {
+  useCallback,
+  useContext,
+  useId,
+  useLayoutEffect,
+  useRef,
+  useState,
+} from 'preact/hooks';
+
+import { useArrowKeyNavigation } from '../../hooks/use-arrow-key-navigation';
+import { useClickAway } from '../../hooks/use-click-away';
+import { useKeyPress } from '../../hooks/use-key-press';
+import { useSyncedRef } from '../../hooks/use-synced-ref';
+import type { PresentationalProps } from '../../types';
+import { MenuCollapseIcon, MenuExpandIcon } from '../icons';
+import Button from './Button';
+import SelectContext from './SelectContext';
+
+export type SelectOptionStatus = {
+  selected: boolean;
+  disabled: boolean;
+};
+
+export type SelectOptionProps<T> = {
+  value: T;
+  disabled?: boolean;
+  children: (status: SelectOptionStatus) => ComponentChildren;
+  classes?: string | string[];
+};
+
+function SelectOption<T>({
+  value,
+  children,
+  disabled = false,
+  classes,
+}: SelectOptionProps<T>) {
+  const selectContext = useContext(SelectContext);
+  if (!selectContext) {
+    throw new Error('Select.Option can only be used as Select child');
+  }
+
+  const { selectValue, value: currentValue } = selectContext;
+  const selected = !disabled && currentValue === value;
+
+  return (
+    <Button
+      variant="custom"
+      classes={classnames(
+        'w-full ring-inset rounded-none !p-0',
+        'border-t first:border-t-0 bg-transparent',
+        { 'hover:bg-grey-1': !disabled },
+        classes,
+      )}
+      onClick={() => selectValue(value)}
+      role="option"
+      disabled={disabled}
+      aria-selected={selected}
+      // This is intended to be focused with arrow keys
+      tabIndex={-1}
+    >
+      <div
+        className={classnames('flex w-full p-1.5 border-l-4', {
+          'border-l-transparent': !selected,
+          'border-l-brand font-medium': selected,
+        })}
+      >
+        {children({ selected, disabled })}
+      </div>
+    </Button>
+  );
+}
+
+export type SelectProps<T> = PresentationalProps & {
+  value: T;
+  onChange: (newValue: T) => void;
+  label: ComponentChildren;
+  disabled?: boolean;
+};
+
+function SelectMain<T>({
+  label,
+  value,
+  onChange,
+  children,
+  disabled,
+  classes,
+  elementRef,
+}: SelectProps<T>) {
+  const [isListboxOpen, setIsListboxOpen] = useState(false);
```

Comment:
```suggestion
  const [listboxOpen, setListboxOpen] = useState(false);
```

I normally only add "is" in the names of the boolean-ness is not obvious enough from the name otherwise.